### ISSUE TITLE
latex-dev: i with macron loops in T1-encoding

### ISSUE THREAD

**Issue by u-fischer** [latex3/latex2e#1587]
In latex-dev a definition for i with macron has been added and that loops in T1-encoding

## Minimal example showing the bug

```latex
\RequirePackage{latexbug}
\documentclass{book}
\usepackage[T1]{fontenc}
%\DeclareTextCompositeCommand{\=}{T1}{\i}{\=\i} %in latex-dev, loops
%\DeclareTextCompositeCommand{\=}{T1}{\i}{something else} %works
\begin{document}
sīc erat scriptum! %similar with \=\i
\end{document}
```

## Log file (required) and possibly PDF file  
log-files is a bit long, but basically one gets

~~~~
\T1\= #1->\@text@composite \T1\= #1\@empty \@text@composite {\add@accent {9}{#1
}}
#1<-\i 

\@text@composite #1#2#3\@text@composite ->\expandafter \@text@composite@x \csna
me \string #1-\string #2\endcsname 
#1<-\T1\= 
#2<-\i 
#3<-\@empty 

\@text@composite@x #1->\ifx #1\relax \expandafter \@secondoftwo \else \expandaf
ter \@firstoftwo \fi #1
#1<-\\T1\=-\i 

\@firstoftwo #1#2->#1
#1<-\\T1\=-\i 
#2<-\add@accent {9}{\i }

\\T1\=-\i ->\=\i 

\=->\T1-cmd \=\T1\= 

\T1-cmd #1->\ifx \protect \@typeset@protect \@inmathwarn #1\else \noexpand #1\e
xpandafter \@gobble \fi 
#1<-\=

\@inmathwarn #1->\ifmmode \@latex@warning {Command \protect #1 invalid in math 
mode}\fi 
#1<-\=

\T1\= #1->\@text@composite \T1\= #1\@empty \@text@composite {\add@accent {9}{#1
}}
#1<-\i 

\@text@composite #1#2#3\@text@composite ->\expandafter \@text@composite@x \csna
me \string #1-\string #2\endcsname 
#1<-\T1\= 
#2<-\i 
#3<-\@empty 

\@text@composite@x #1->\ifx #1\relax \expandafter \@secondoftwo \else \expandaf
ter \@firstoftwo \fi #1
#1<-\\T1\=-\i 

\@firstoftwo #1#2->#1
#1<-\\T1\=-\i 
#2<-\add@accent {9}{\i }

\\T1\=-\i ->\=\i 

\=->\T1-cmd \=\T1\= 

\T1-cmd #1->\ifx \protect \@typeset@protect \@inmathwarn #1\else \noexpand #1\e
xpandafter \@gobble \fi 
#1<-\=

\@inmathwarn #1->\ifmmode \@latex@warning {Command \protect #1 invalid in math 
mode}\fi 
#1<-\=

\T1\= #1->\@text@composite \T1\= #1\@empty \@text@composite {\add@accent {9}{#1
}}
#1<-\i 
~~~~

**Comment by cfr42** [comment]
`\DeclareTextCompositeCommand{\=}{T1}{\i}{\={\char25}}`

**Comment by FrankMittelbach** [comment]
> `\DeclareTextCompositeCommand{\=}{T1}{\i}{\={\char25}}`

yes, that'l probably do, but why do it in the first place (it works correctly without any declaration afaics)? 

I think what was intended was
```
\DeclareTextCompositeCommand{\=}{T1}{i}{\=\i}
```
(but I didn't find a reference why the change was made)

**Comment by josephwright** [comment]
@FrankMittelbach https://github.com/latex3/latex2e/issues/1453

**Comment by josephwright** [comment]
Hmm, I think I meant
```latex
\DeclareTextCompositeCommand{\=}{T1}{i}{\=\i}
```
Does that look right?

**Comment by FrankMittelbach** [comment]
So it is as I thought:  the change has a `\` too many  (the changelog omits the link to that issue) 

but then the question is wouldn a similar definition be needed for `\=j` or is this something that doesn't exist in any spoken or written language?

**Comment by FrankMittelbach** [comment]
> Hmm, I think I meant
> 
> ```
> \DeclareTextCompositeCommand{\=}{T1}{i}{\=\i}
> ```
> 
> Does that look right?

yes, and please add the (gh/1453) in the changes logs

**Comment by josephwright** [comment]
> > Hmm, I think I meant
> > ```
> > \DeclareTextCompositeCommand{\=}{T1}{i}{\=\i}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Does that look right?
> 
> yes, and please add the (gh/1453) in the changes logs

OK - I always use the Git blame for changes so to me there is a clear link :)

**Comment by car222222** [comment]
@FrankMittelbach wrote:
 needed for \=j 
or is this something that doesn't exist in any spoken or written language?

Very likely not needed, but I could invent a new language that does, if you like! 

Do we have any "accented j" characters/glyphs so far? 

Check Unicode, maybe! ??

**Comment by davidcarlisle** [comment]
On Sat, 21 Dec 2024 at 09:19, Chris Rowley ***@***.***> wrote:

> @FrankMittelbach <https://github.com/FrankMittelbach> wrote:
> needed for =j
> or is this something that doesn't exist in any spoken or written language?
>
> Very likely not needed, but I could invent a new language that does, if
> you like!
>
> Do we have any "accented j" characters/glyphs so far?
>
> Check Unicode, maybe! ??
>

U+0135  is j with circumflex
U+01f0 is j with caron

(tuenc.def supports all four of  \^j, \^\j, \v{j} and \v\j  as composites
already)

**Comment by FrankMittelbach** [comment]
> > > 
> > 
> > 
> > yes, and please add the (gh/1453) in the changes logs
> 
> OK - I always use the Git blame for changes so to me there is a clear link :)

as far as I'm concerned that is a rather lengthly extra step, compared to just go to /issue/number and I'm not sure that blame shows me the releated discussions necessarily (does it? I think I only saw that you made the change but not why)
